plugins/woocommerce-admin/client/core-profiler/index.tsx

```diff
@@ -0,0 +1,337 @@
+/**
+ * External dependencies
+ */
+import { createMachine, assign } from 'xstate';
+import { useMachine } from '@xstate/react';
+import { useEffect } from '@wordpress/element';
+import { ExtensionList } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import { IntroOptIn } from './pages/IntroOptIn';
+import { UserProfile } from './pages/UserProfile';
+import { BusinessInfo } from './pages/BusinessInfo';
+import { BusinessLocation } from './pages/BusinessLocation';
+
+/** Uncomment below to display xstate inspector during development */
```

Comment:
> Improve the XState devtools to be toggled on and off automatically by detecting environment

I'd love to see that. 🙌  I think one way is to use `new webpack.DefinePlugin` to pass the environment variable.

Reply:
Thanks for the suggestion! I didn't have any idea in mind so I left it as a todo so that I could get this PR out ASAP 😆